odata-client-runtime/src/main/java/com/github/davidmoten/odata/client/Serializer.java

```diff
@@ -128,14 +129,22 @@ public String serialize(Object entity) {
             JsonNode tree = m.readTree(s);
             ObjectNode o = (ObjectNode) tree;
             ChangedFields cf = entity.getChangedFields();
+            Set<String> cfs = cf.toSet();
             List<String> list = new ArrayList<>();
             Iterator<String> it = o.fieldNames();
             while (it.hasNext()) {
                 String name = it.next();
                 if (!cf.contains(name) && !name.equals("@odata.type")) {
                     list.add(name);
                 }
+                cfs.remove(name);
             }
+            // Make sure we add any changed fields that don't appear in the tree (i.e. the null ones)
```

Comment:
I don't get this one yet. This needs motivation. What is not working in the API?

Reply:
If you want to remove an attribute from a target system, you need to send a patch request containing a null value for that attribute (e.g. {"attribute": null}).  Currently these are filtered out by the serialiser because null attributes don't appear in the list of attributes under consideration for the patch request despite being in the list of changed attributes.

Reply:
It's a good catch then, I thought there was testing at the API level somewhere for this. I'll have a look. Can you add a test please (that fails on the current code).

Reply:
I've added a test case for the null-value patch operations

Reply:
Thanks for the test. We don't need the new code in `serializeChangesOnly`, we just need to ensure the `Include.NON_NULLS` option is not set in the ObjectMapper used by that method.

Modify the ObjectMapper code and change `serializerChangesOnly` like this:

```java
    private static ObjectMapper createObjectMapper() {
        return createObjectMapper(false);
    }
    
    private static ObjectMapper createObjectMapper(boolean includeNulls) {
        return new ObjectMapper() //
                .registerModule(new Jdk8Module()) //
                .registerModule(new JavaTimeModule())
                .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false) //
                .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS) //
                .setSerializationInclusion(includeNulls? Include.ALWAYS : Include.NON_NULL) //
                // StdDateFormat is ISO8601 since jackson 2.9
                .setDateFormat(new StdDateFormat().withColonInTimeZone(true));
    }
...
    public <T extends ODataEntityType> String serializeChangesOnly(T entity) {

            ObjectMapper m = createObjectMapper(true);	           
```
Note that I've made createObjectMapper private as well.